pytorch_lightning/callbacks/gpu_stats_monitor.py

```diff
@@ -0,0 +1,164 @@
+            at the start and end of each step. Default: ``True``.
+        intra_step_time: Set to ``True`` to log the time of each step. Default: ``False``.
+        inter_step_time: Set to ``True`` to log the time between the end of one step
+            and the start of the next step. Default: ``False``.
+        fan_speed: Set to ``True`` to log percentage of fan speed. Default: ``False``.
+        temperature: Set to ``True`` to log the memory and gpu temperature in degree Celsius.
+            Default: ``False``.
+
+    Example::
+
+        >>> from pytorch_lightning import Trainer
+        >>> from pytorch_lightning.callbacks import GPUStatsMonitor
+        >>> gpu_stats = GPUStatsMonitor() # doctest: +SKIP
+        >>> trainer = Trainer(callbacks=[gpu_stats]) # doctest: +SKIP
+
+    GPU stats are mainly based on `nvidia-smi --query-gpu` command. The description of the queries is as follows:
+
+    - **fan.speed** – The fan speed value is the percent of maximum speed that the device's fan is currently
+      intended to run at. It ranges from 0 to 100 %. Note: The reported speed is the intended fan speed.
+      If the fan is physically blocked and unable to spin, this output will not match the actual fan speed.
+      Many parts do not report fan speeds because they rely on cooling via fans in the surrounding enclosure.
+    - **memory.used** – Total memory allocated by active contexts.
+    - **memory.free** – Total free memory.
+    - **utilization.gpu** – Percent of time over the past sample period during which one or more kernels was
+      executing on the GPU. The sample period may be between 1 second and 1/6 second depending on the product.
+    - **utilization.memory** – Percent of time over the past sample period during which global (device) memory was
+      being read or written. The sample period may be between 1 second and 1/6 second depending on the product.
+    - **temperature.gpu** – Core GPU temperature, in degrees C.
+    - **temperature.memory** – HBM memory temperature, in degrees C.
+
+    """
+
+    def __init__(
+        self,
+        memory_utilization: bool = True,
+        gpu_utilization: bool = True,
+        intra_step_time: bool = False,
+        inter_step_time: bool = False,
+        fan_speed: bool = False,
+        temperature: bool = False
+    ):
+        super().__init__()
+
+        if shutil.which("nvidia-smi") is None:
+            raise MisconfigurationException(
+                'Cannot use GPUStatsMonitor callback because NVIDIA driver is not installed.'
+            )
+
+        self._log_stats = AttributeDict({
+            'memory_utilization': memory_utilization,
+            'gpu_utilization': gpu_utilization,
+            'intra_step_time': intra_step_time,
+            'inter_step_time': inter_step_time,
+            'fan_speed': fan_speed,
+            'temperature': temperature
+        })
+
+    def on_train_batch_start(self, trainer, pl_module, batch, batch_idx, dataloader_idx):
```

Comment:
```suggestion
    @rank_zero_only
    def on_train_batch_start(self, trainer, pl_module, batch, batch_idx, dataloader_idx):
```
can we add this to every callback method?